### PR TITLE
[RPD-225] Move matcha.state file from `infrastructure/resources/` to `infrastructure/`

### PR DESCRIPTION
**Pull request by Christopher-Norman**
This PR moves `matcha.state` from `infrastructure/resources/` to `infrastructure/`

## Checklist

Please ensure you have done the following:

* [x] I have read the [CONTRIBUTING](https://github.com/fuzzylabs/matcha/blob/main/CONTRIBUTING.md) guide.
* [ ] I have updated the documentation if required.
* [x] I have added tests which cover my changes.

## Type of change

Tick all those that apply:

* [ ] Bug Fix (non-breaking change, fixing an issue)
* [ ] New feature (non-breaking change to add functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Other (add details above)

### CODE REVIEW

File: src/matcha_ml/services/terraform_service.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ class TerraformConfig:
     @property
     def state_file(self) -> str:
         """The path to the state file."""
-        return os.path.join(self.working_dir, "matcha.state")
+        return os.path.join(self.working_dir, "..", "matcha.state")
```
Comment (dudeperf3ct) [inline src/matcha_ml/services/terraform_service.py:24]: `..` can be repalced by `os..path.pardir`. 

> `os.path.pardir` stands for ".." which means one directory above the current one.

Also, please run everything after making this change to be sure nothing is breaking and `matcha.state` file is created in correct place.

File: src/matcha_ml/templates/azure_template/azure_template.py [hunk 2]
```diff
@@ -88,7 +88,8 @@ def build_template(
         """
         super().build_template(config, template_src, destination, verbose)
 
-        state_file_destination = os.path.join(destination, "matcha.state")
+        # Add matcha.state file one directory above the template
+        state_file_destination = os.path.join(destination, "..", "matcha.state")
```
Comment (dudeperf3ct) [inline src/matcha_ml/templates/azure_template/azure_template.py:92]: Same note as above about `..`.

File: tests/test_cli/test_provision.py [hunk 3]
```diff
@@ -78,7 +78,7 @@ def assert_infrastructure(
 
     if check_matcha_state_file:
         # Check that matcha state file exists and content is equal/correct
-        state_file_path = os.path.join(destination_path, "matcha.state")
+        state_file_path = os.path.join(destination_path, "..", "matcha.state")
```
Comment (dudeperf3ct) [inline tests/test_cli/test_provision.py:81]: Same note as above.

**Review by dudeperf3ct — CHANGES_REQUESTED** [review]

File: src/matcha_ml/state/matcha_state.py [hunk 4]
```diff
@@ -7,9 +7,7 @@
 class MatchaStateService:
     """A matcha state service for handling to matcha.state file."""
 
-    matcha_state_dir = os.path.join(
-        ".matcha", "infrastructure", "resources", "matcha.state"
-    )
+    matcha_state_dir = os.path.join(".matcha", "infrastructure", "matcha.state")
```
Comment (dudeperf3ct) [inline src/matcha_ml/state/matcha_state.py:10]: Will this change from PR 124?

**Review by dudeperf3ct — APPROVED** [review]

File: src/matcha_ml/state/matcha_state.py (hunk 4 above)
Comment (Christopher-Norman) [reply src/matcha_ml/state/matcha_state.py:10]: Depending on the order merged there will be a conflict, but yes both the name and the path will be updated.

**Review by swells2020 — APPROVED** [review]
Just the one though RE using the StateService to yield the matcha.state filepath rather than hardcoding, but otherwise looks great!

File: src/matcha_ml/templates/azure_template/azure_template.py [hunk 5]
```diff
@@ -88,7 +88,8 @@ def build_template(
         """
         super().build_template(config, template_src, destination, verbose)
 
-        state_file_destination = os.path.join(destination, "matcha.state")
+        # Add matcha.state file one directory above the template
+        state_file_destination = os.path.join(destination, os.pardir, "matcha.state")
```
[the remark this answers is not in the document]
Comment (Christopher-Norman) [reply src/matcha_ml/templates/azure_template/azure_template.py:92]: For this function, we cannot as we pass `destination` as a parameter.

**Pull request merged**